ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -229,5 +366,35 @@ long getCapacityInBytes() {
     List<String> getLogSegmentsNotInJournal() throws StoreException {
       return validLogSegments;
     }
+
+    @Override
+    void compact(CompactionDetails details) {
+      compactCalled = true;
+      compactCallsCountdown.countDown();
+      if (details == null) {
+        callOrderException = new Exception("Called compact() with null details");
+      } else if (!resumeCompactionCalled) {
+        callOrderException = new Exception("Called compact() before calling maybeResumeCompaction()");
```

Comment:
I don't think this is mimicing actual production code. Can you show me where exactly(in prod code) are we checking that compact() is not called before mayBeResumeCompaction() (if not exception is thrown)?